pycalphad/variables.py

```diff
@@ -244,10 +244,12 @@ def __init__(self, phase_name, subl_index, species): #pylint: disable=W0221
             self._self_without_suffix = self
 
     def compute_property(self, compsets, cur_conds, chemical_potentials):
-        state_variables = compsets[0].phase_record.state_variables
         result = np.atleast_1d(np.full(self.shape, fill_value=np.nan))
+        state_variables = None
         for _, compset in self.filtered(compsets):
             # we'll only hit this code path if this phase is present in the list of compsets
+            if state_variables is None:
+                state_variables = compsets[0].phase_record.state_variables
```

Comment:
```suggestion
                state_variables = compset.phase_record.state_variables
```

Reply:
`self.filtered(compsets)` may not include `compsets[0]` (though in principle the state variables should all agree)

Reply:
do we expect that `compset[0]` could have more correct state variables? Isn't that already applying the assumption that the state variables are the same across all the compsets? Since this is in `SiteFraction`, it's possible that `compset[0]` isn't even the phase we're interested in.

Reply:
I don't think it should matter. I just wanted to raise it to be sure.